File: infinity-server/src/main/java/io/github/derechtepilz/infinity/gamemode/modification/PortalDisableHandler.java
package io.github.derechtepilz.infinity.gamemode.modification;

import io.github.derechtepilz.infinity.gamemode.Gamemode;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.world.PortalCreateEvent;

public class PortalDisableHandler implements Listener {

	@EventHandler
	public void onCreatePortal(PortalCreateEvent event) {
		if (Gamemode.getFromKey(event.getWorld().getKey()) == Gamemode.INFINITY) {
			event.setCancelled(true);
		}
	}

}

File: src/main/kotlin/io/github/derechtepilz/infinity/gamemode/PortalDisableHandler.kt
package io.github.derechtepilz.infinity.gamemode

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.world.PortalCreateEvent

class PortalDisableHandler : Listener {

	@EventHandler
	fun onCreatePortal(event: PortalCreateEvent) {
		if (Gamemode.getFromKey(event.world.key) == Gamemode.INFINITY) {
			event.isCancelled = true
		}
	}

}